DataFormats/PatCandidates/interface/PackedTriggerPrescales.h

```diff
@@ -14,9 +14,9 @@ namespace pat {
     ~PackedTriggerPrescales() {}
 
     // get prescale by index.
-    int getPrescaleForIndex(int index) const;
+    float getPrescaleForIndex(int index) const;
     // get prescale by name or name prefix (if setTriggerNames was called)
-    int getPrescaleForName(const std::string &name, bool prefixOnly = false) const;
+    float getPrescaleForName(const std::string &name, bool prefixOnly = false) const;
```

Comment:
given that this change can generate wrong results in client code, can it be changed as to break anyone using ```int``` as the return type at compile time? (which perhaps can only be accomplished by a name change)

Reply:
is there a way to enable `-Wconversion` for code that calls this function ?

Reply:
I did not find a way to attach it to the function, but we could enable it for all code that includes this header, adding
```c++
#pragma GCC diagnostic error "-Wconversion"
```

towards the top, after the `#include`s.

Reply:
(we could also consider enabling it globally for all CMSSW, but I guess we will have plenty of positives...)

Reply:
Adding #pragma GCC diagnostic error "-Wconversion" introduces errors from unrelated files included in the headers so I could not add this line

Reply:
Makes sense. Let's see if we can eventually add it globally: https://github.com/cms-sw/cmssw/issues/39098 .